docs/config/@config.md

```diff
@@ -0,0 +1,219 @@
+`@Config`
+=======
+
+The `@Config` annotation is an alternative to `Configuration`. 
+
+Table of Contents
+-------------
+
+* [Basics][basics]
+* [@Config Use][configuse]
+* [@Comment Use][commentuse]
+* [@Name Use][nameuse]
+* [@RangeInt Use][rangeintuse]
+* [@RangeDouble Use][rangedoubleuse]
+* [@LangKey Use][langkeyuse]
+* [@RequiresMcRestart Use][requiresmcrestartuse]
+* [@RequiresWorldRestart Use][requiresworldrestartuse]
+
+Basics
+------
+
+A class annotated with `@Config` will have any fields turned into config options. Said fields can be annotated to add information, using the plethora of annotations provided in the `@Config` class.
+
+!!! note
+
+    Enum config values generate a comment that is as many lines long as the enum has values.
+
+A good example of this system is [`Forge's Test`][forgetest]
+
+@Config Use
+-----------
+
+This annotation is used to denote a class is the container for some configuration options. 
+
+There are 4 properties:
+
+|     Property|Type     | Default Value | Comment 
+|           -:|:-       |     :-:       |:-
+|        modid|`String` |     N/A       | The mod id that this configuration is associated with. 
+|         name|`String` |     `""`      | A user friendly name for the config file, the default will be modid. 
+|         type|`Type`   |`Type.INSTANCE`| The type this is, right now the only value is `Type.INSTANCE`. This is intended to be expanded upon later for more Forge controlled configs. 
+|     category|`String` |  `"general"`  | Root element category. If this is an empty string then the root category is disabled. 
+
+!!! important
+
+    If you disable the root category, this will cause issues unless you create [Sub Categories][subcategories].
+
+@Comment Use
+------------
+
+Adding a `@Comment` annotation to a field will add a comment to the config in its file. 
+
+It has 1 property:
+
+| Property|Type       | Default Value |Comment 
+|       -:|:-         |       :-:     |:-
+|    value|`String[]` |       N/A     |A `String[]` will form a multi-line comment where 1 element = 1 line.
+
+!!! note
+
+    The value in this annotation can be passed as a `String` because java is cool and turns it into a `String[]`.
+
+### Example
+```java
+@Comment({
+  "You can add comments using this",
+  "and if you supply an array it will be multi-line"
+})
+public static boolean doTheThing = true;
+```
+This would produce the following config:
+```
+# You can add comments using this
+# and if you supply an array it will me multiline
```

Comment:
This comment does not match the value provided in the code example. Specifically, `multiline` should be `multi-line`, but I would just copy them directly to make sure there are no other typos.